authapi/tests.py

```diff
@@ -1,7 +1,339 @@
+from django.contrib.auth.models import User
+from django.core.urlresolvers import reverse
+from rest_framework import status
+from rest_framework.request import Request
+from rest_framework.reverse import reverse as drt_reverse
+from rest_framework.test import APITestCase, APIRequestFactory
 
+from authapi.serializers import (
+    UserSerializer, TeamSerializer, OrganizationSummarySerializer,
+    TeamSummarySerializer, PermissionSerializer, UserSummarySerializer,
+    OrganizationSerializer)
+from authapi.models import SeedTeam, SeedOrganization, SeedPermission
 
-class RunningTests(TestCase):
-    def test_tests_are_run(self):
-        '''Make sure that our tests are actually running.'''
-        pass
+
+class AuthAPITestCase(APITestCase):
+    def get_context(self, url):
+        '''Returns the request context for a given url.'''
+        factory = APIRequestFactory()
+        request = factory.get(url)
+        return {
+            'request': Request(request)
+        }
+
+    def get_full_url(self, viewname, *args, **kwargs):
+        '''Returns the full URL, with host and port. Takes the same arguments
+        as reverse.'''
+        factory = APIRequestFactory()
+        part_url = reverse(viewname, *args, **kwargs)
+        request = factory.get(part_url)
+        kwargs['request'] = request
+        return drt_reverse(viewname, *args, **kwargs)
+
+
+class UserTests(AuthAPITestCase):
+    def test_get_account_list_empty(self):
+        '''If there are no accounts, and empty list should be returned.'''
+        response = self.client.get(reverse('user-list'))
+        self.assertEqual(response.data, [])
+
+    def test_get_account_list_multiple(self):
+        '''If there are multiple users, it should return them all in a list.'''
+        user1 = User.objects.create_user(username="user1@example.org")
+        user2 = User.objects.create_user(username="user2@example.org")
+
+        context = self.get_context(reverse('user-list'))
+        expected = [
+            UserSerializer(instance=u, context=context).data
+            for u in [user1, user2]
+        ]
+
+        response = self.client.get(reverse('user-list'))
+        self.assertEqual(
+            sorted(expected, key=lambda i: i['id']),
+            sorted(response.data, key=lambda i: i['id']))
+
+    def test_create_user_no_required_fields(self):
+        '''A POST request to the user endpoint should return an error if there
+        is no email field, as it is required.'''
+        response = self.client.post(reverse('user-list'), data={})
+        self.assertEqual(response.data, {
+            'email': ['This field is required.']
+        })
+
+    def test_create_user(self):
+        '''A POST request to the user endpoint should create a user with all
+        of the supplied details.'''
+        data = {
+            'email': 'user1@example.org',
+            'first_name': 'user1',
+            'last_name': 'example',
+            'admin': True,
+        }
```

Comment:
So, password is an interesting one. What we really need to be doing is called `create_user` and `create_superuser` instead of `create`, that way the password will be turned into a hash and stored in the `password` field, instead of plain-text. I was planning on doing all of this in the PR where we actually implement users. This skeleton was more focused on getting the framework up than us actually creating objects through the API.